### PR TITLE
feat(misconf): Add terraformplan support

### PR DESCRIPTION
**Pull request by simar7**
## Description
Adds support to scan terraformplan files.

## Related issues
- Close: https://github.com/aquasecurity/trivy/issues/4341

## Related PRs
- [x] https://github.com/aquasecurity/defsec/pull/1320

## Checklist
- [ ] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [ ] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

**Comment by simar7** [comment]
@knqyf263 - I've updated the tests and rebased. Could you take a look?

File: go.mod [hunk 1]
```diff
@@ -14,7 +14,7 @@ require (
 	github.com/NYTimes/gziphandler v1.1.1
 	github.com/alicebob/miniredis/v2 v2.30.2
 	github.com/aquasecurity/bolt-fixtures v0.0.0-20200903104109-d34e7f983986
-	github.com/aquasecurity/defsec v0.88.1
+	github.com/aquasecurity/defsec v0.88.2-0.20230516215146-673ff3afe374
```
Comment (simar7) [inline go.mod:17]: note: update to latest when https://github.com/aquasecurity/trivy/pull/4474 is merged

**Review by knqyf263 — APPROVED** [review]
LGTM. We need to resolve a conflict, though.

**Comment by simar7** [comment]
> LGTM. We need to resolve a conflict, though.

Will fix that. Need this PR to be merged first https://github.com/aquasecurity/trivy/pull/4474 and then I'll rebase this one.

**Comment by simar7** [comment]
> > LGTM. We need to resolve a conflict, though.
> 
> Will fix that. Need this PR to be merged first #4474 and then I'll rebase this one.

I realized its easier to rebase the other one on top of this. I'll merge this and rebase the other.

**Pull request merged**